pkg/controller/nodes/task/handler.go

```diff
@@ -460,6 +476,13 @@ func (t Handler) invokePlugin(ctx context.Context, p pluginCore.Plugin, tCtx *ta
 		} else {
 			pluginTrns.ObserveSuccess(tCtx.ow.GetOutputPath(), &event.TaskNodeMetadata{CacheStatus: cacheStatus.GetCacheStatus(), CatalogKey: cacheStatus.GetMetadata()})
 		}
+
+		ownerID := tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName()
+		reservation, err := t.ReleaseCatalogReservation(ctx, ownerID, tCtx.tr, tCtx.InputReader())
```

Comment:
ahh i see it, but should we do this in finalize?

Reply:
one problem is today we optimize finalize as it is a discrete step, we optimize it and use only for non pod tasks

Reply:
done. this makes perfect sense, in addition to failures it releases the reservations on aborts as well.